apps/studio/components/to-be-cleaned/Storage/StorageSettings/S3Connection.tsx

```diff
@@ -0,0 +1,341 @@
+                    setShowSuccess(true)
+                  }}
+                >
+                  <Label_Shadcn_ htmlFor="description">Description</Label_Shadcn_>
+                  <Input
+                    autoComplete="off"
+                    placeholder="Key description"
+                    type="text"
+                    name="description"
+                    required
+                  />
+                  <div className="flex justify-end">
+                    <Button
+                      className="mt-4"
+                      htmlType="submit"
+                      loading={createStorageCreds.isLoading}
+                    >
+                      Create credential
+                    </Button>
+                  </div>
+                </form>
+              </>
+            )}
+          </DialogContent>
+        </Dialog>
+      </div>
+
+      <div
+        className={cn([
+          'bg-surface-100',
+          'overflow-hidden border-muted',
+          'rounded-md border shadow mt-6',
+        ])}
+      >
+        {storageCredsQuery.isLoading ? (
+          <div className="p-4">
+            <GenericSkeletonLoader />
+          </div>
+        ) : (
+          <div className="overflow-x-auto">
+            <Table
+              className=""
+              head={[
+                <Table.th key="">Description</Table.th>,
+                <Table.th key="">Access key</Table.th>,
+                <Table.th key="">Created at</Table.th>,
+                <Table.th key="actions" />,
+              ]}
+              body={
+                hasStorageCreds ? (
+                  storageCreds.data?.map((cred: any) => (
+                    <StorageCredItem
+                      key={cred.id}
+                      created_at={cred.created_at}
+                      access_key={cred.access_key}
+                      description={cred.description}
+                      id={cred.id}
+                      onDeleteClick={() => {
+                        setDeleteCredId(cred.id)
+                        setOpenDeleteDialog(true)
+                      }}
+                    />
+                  ))
+                ) : (
+                  <Table.tr>
+                    <Table.td colSpan={4}>
+                      <p className="text-sm text-foreground">No credentials created</p>
+                      <p className="text-sm text-foreground-light">
+                        There are no credentials associated with your project yet
+                      </p>
+                    </Table.td>
+                  </Table.tr>
+                )
+              }
+            />
+          </div>
+        )}
+      </div>
+
+      <Dialog open={openDeleteDialog} onOpenChange={setOpenDeleteDialog}>
+        <DialogContent className="p-4">
+          <DialogTitle>Revoke storage credentials</DialogTitle>
+          <DialogDescription>
+            This action will permanently revoke the credentials. Requests made with these
```

Comment:
Just "This action is irreversible and requests made with these credentials will stop working." is enough in the body imo. we have 4 words of revoke in the current copy haha
![CleanShot 2024-04-12@2x](https://github.com/supabase/supabase/assets/2155545/11a56085-0000-4c45-9d5d-55858e77e32c)